dev-academy/develop-smart-contract/02-develop.md

```diff
@@ -176,211 +146,483 @@ pub fn execute(
     msg: ExecuteMsg,
 ) -> Result<Response, ContractError> {
     match msg {
-        ExecuteMsg::CreatePot {
-            target_addr,
-            threshold,
-        } => execute_create_pot(deps, info, target_addr, threshold),
-        ExecuteMsg::Receive(msg) => execute_receive(deps, info, msg),
+        ExecuteMsg::NewEntry {description, priority} => create_new_entry(deps, info, description, priority),
+        ExecuteMsg::UpdateEntry {id, description, status, priority } => update_entry(deps, info, id, description, status, priority),
+        ExecuteMsg::DeleteEntry {id} => delete_entry(deps, info, id)
     }
 }
 ```
 
-The example above is a great one to understand the [match](https://doc.rust-lang.org/rust-by-example/flow_control/match.html) operator.
-Now that we have added additional options to `ExecuteMsg` we will go back and update the enum.
+The function `create_new_entry()` handles the creation of a new entry on the list. 
+* Before creating the new entry, the function checks if the message sender is the owner of the contract. If not, it returns an error and the new entry creation fails to be performed.
```

Comment:
For these steps better to use numbering, it indicates this is a ordered algorithm
```suggestion
1. Before creating the new entry, the function checks if the message sender is the owner of the contract. If not, it returns an error and the new entry creation fails to be performed.
```